librubyfmt/src/format.rs

```diff
@@ -852,42 +852,57 @@ pub fn format_assocs(
 ) {
     let len = assocs.len();
     for (idx, assoc) in assocs.into_iter().enumerate() {
-        if sc != SpecialCase::NoLeadingTrailingCollectionMarkers || idx != 0 {
-            ps.emit_soft_indent();
-        }
-        ps.with_start_of_line(
-            false,
-            Box::new(|ps| match assoc {
-                AssocNewOrAssocSplat::AssocNew(new) => {
-                    match new.1 {
-                        AssocKey::Label(label) => {
-                            handle_string_and_linecol(ps, label.1, label.2);
-                            ps.emit_space();
-                        }
-                        AssocKey::Expression(expression) => {
-                            format_expression(ps, expression);
-                            ps.emit_space();
-                            ps.emit_ident("=>".to_string());
-                            ps.emit_space();
-                        }
-                    }
-                    format_expression(ps, new.2);
-                }
-                AssocNewOrAssocSplat::AssocSplat(splat) => {
-                    ps.emit_ident("**".to_string());
-                    format_expression(ps, splat.1);
-                }
-            }),
-        );
+        ps.emit_soft_indent();
```

Comment:
does this need the special case?

Reply:
No, since that was only necessary for one scenario I pushed that to the caller to remove a conditional: https://github.com/penelopezone/rubyfmt/pull/285/files#diff-63b92541cacb647f9227c6eac250e906cb60dade9a39a892517f2d99ec122a28R694